src/Symfony/Component/Form/MissingDataHandler.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form;
+
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+
+class MissingDataHandler
+{
+    public readonly \stdClass $missingData;
+
+    public function __construct()
+    {
+        $this->missingData = new \stdClass();
+    }
+
+    public function handle(FormInterface $form, mixed $data): mixed
+    {
+        $processedData = $this->handleMissingData($form, $data);
+
+        return $processedData === $this->missingData ? $data : $processedData;
+    }
+
+    private function handleMissingData(FormInterface $form, mixed $data): mixed
+    {
+        try {
+            if ($form->getConfig()->getType() instanceof ResolvedFormTypeInterface && $form->getConfig()->getType()->getInnerType() instanceof CheckboxType) {
+                $falseValues = $form->getConfig()->getOption('false_values');
+
+                if ($data === $this->missingData) {
+                    return $falseValues[0];
+                }
+
+                if (\in_array($data, $falseValues)) {
+                    return $data;
+                }
+            }
+        } catch (\Error $error) {
```

Comment:
Catching type errors here doesn't look like the right solution to me. We should investigate instead why the `$type` property in `FormConfigBuilder` isn't initialized at this point and fix that root issue instead.

Reply:
I don't like it either. 

Another solution could be making FormConfigBuilderInterface::getType() nullable. Then null state would have to be handled in these methods where getType() is called:

ValidatorDataCollector#getCasters
FormDataExtractor#extractConfiguration
FormDataCollector#getCasters
FormTypeCsrfExtension#finishView
FormTypeCsrfExtension#buildForm
BaseType#buildView
Button#createView

I don't know if null/undefined type is a valid case or not.

Reply:
Thank you. When I created this PR on Jan 19 it worked. These commits looks related: https://github.com/symfony/symfony/commit/a53b450d https://github.com/symfony/symfony/commit/488a262a